src/main/java/maksym/fedorenko/bookstore/exception/GlobalExceptionHandler.java

```diff
@@ -0,0 +1,67 @@
+package maksym.fedorenko.bookstore.exception;
+
+import java.time.LocalDateTime;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class GlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex, HttpHeaders headers,
+            HttpStatusCode status, WebRequest request
+    ) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", HttpStatus.BAD_REQUEST);
+        List<String> errors = ex.getBindingResult().getAllErrors().stream()
+                .map(this::getErrorMessage)
+                .toList();
+        body.put("errors", errors);
+        return new ResponseEntity<>(body, headers, status);
+    }
+
+    @ExceptionHandler(value = EntityNotFoundException.class)
+    protected ResponseEntity<Object> handleEntityNotFoundException(RuntimeException ex) {
+        HttpStatus status = HttpStatus.NOT_FOUND;
+        return getResponseEntity(ex, status);
+    }
+
+    @ExceptionHandler(value = EntityNotSavedException.class)
+    protected ResponseEntity<Object> handleEntityNotSavedException(RuntimeException ex) {
+        return getResponseEntity(ex, HttpStatus.BAD_REQUEST);
+    }
+
+    @ExceptionHandler(DataIntegrityViolationException.class)
+    public ResponseEntity<Object> handleConflict(RuntimeException ex) {
+        return getResponseEntity(ex, HttpStatus.CONFLICT);
+    }
+
+    private ResponseEntity<Object> getResponseEntity(RuntimeException ex, HttpStatus status) {
+        Map<String, Object> body = new LinkedHashMap<>();
+        body.put("timestamp", LocalDateTime.now());
+        body.put("status", status);
+        body.put("errors", ex.getMessage());
+        return new ResponseEntity<>(body, status);
+    }
```

Comment:
overall this is a bad error handling :)

don't try to create a single method for all the possible cases

here's my proposal:

1. you create a model for representing errors. it can be something like:
```
public record ErrorResponseWrapper(String error, String details) {}
```

2. you use that wrapper for each particular handler like:
```
@ExceptionHandler(value = EntityNotSavedException.class)
@ResponseStatus(HttpStatus...) // pick up an appropriate status
 public ErrorResponseWrapper handleEntityNotSavedException(EntityNotSavedException ex) {
    return new ErrorResponseWrapper("entity-not-saved", ex.getMessage());
}
```
and each handler will have a unique error and status

3. you annotate your class as `@RestControllerAdvice`, so that your wrapper gets serialized in JSON

voila :) you have a pretty fine error handling


**IMPORTANT**
the approach I described above is mostly for your custom exceptions (although you can easily used other ones)

you may also use `ResponseEntityExceptionHandler` for handling exceptions, predefined by Spring - in your case it is `MethodArgumentNotValidException`

these 2 approaches work well together